doc/source/dev/conduct/code_of_conduct.rst

```diff
@@ -0,0 +1,185 @@
+   are not acceptable. This includes, but is not limited to:
+
+    - Violent threats or language directed against another person.
+    - Sexist, racist, or otherwise discriminatory jokes and language.
+    - Posting sexually explicit or violent material.
+    - Posting (or threatening to post) other people's personally identifying information ("doxing").
+    - Sharing private content, such as emails sent privately or non-publicly, or unlogged forums such as IRC channel history.
+    - Personal insults, especially those using racist or sexist terms.
+    - Unwelcome sexual attention.
+    - Excessive or unnecessary profanity.
+    - Repeated harassment of others. In general, if someone asks you to stop, then stop.
+    - Advocating for, or encouraging, any of the above behaviour.
+
+5. Be concise. Keep in mind that what you write once will be read by hundreds
+   of persons. Writing a short email means people can understand the
+   conversation as efficiently as possible. Short emails should always strive
+   to be empathetic, welcoming, friendly and patient. When a long explanation
+   is necessary, consider adding a summary.
+
+6. Try to bring new ideas to a conversation so that each mail adds something
+   unique to the thread, keeping in mind that the rest of the thread still
+   contains the other messages with arguments that have already been made.
+
+7. Try to stay on topic, especially in discussions that are already fairly
+   large.
+
+8. Step down considerately. Members of every project come and go. When somebody
+   leaves or disengages from the project they should tell people they are
+   leaving and take the proper steps to ensure that others can pick up where
+   they left off. In doing so, they should remain respectful of those who
+   continue to participate in the project and should not misrepresent the
+   project's goals or achievements. Likewise, community members should respect
+   any individual's choice to leave the project.
+
+
+Diversity Statement
+-------------------
+
+The SciPy project welcomes and encourages participation by everyone. We are
+committed to being a community that everyone feels good about joining. Although
+we may not be able to satisfy everyone, we will always work to treat everyone
+well.
+
+No matter how you identify yourself or how others perceive you: we welcome you.
+Though no list can hope to be comprehensive, we explicitly honour diversity in:
+age, culture, ethnicity, genotype, gender identity or expression, language,
+national origin, neurotype, phenotype, political beliefs, profession, race,
+religion, sexual orientation, socioeconomic status, subculture and technical
+ability.
+
+Though we welcome people fluent in all languages, SciPy development is
+conducted in English.
+
+Standards for behaviour in the SciPy community are detailed in the Code of
+Conduct above. We expect participants in our community to meet these standards
+in all their interactions and to help others to do so as well.
```

Comment:
Add a forward reference here, so that readers know where to find guidelines on helping others?

Reply:
I'm not sure there's a good place to reference. Or did you mean the next section?

Reply:
Yes, I meant a pointer to the next section, so that readers aren't left wondering "how do I meet these standards in my interactions".  A simple "see below" should suffice.